extentions/gcc.bzl

```
@@ -21,9 +21,76 @@ def _gcc_impl(mctx):
         if not mod.is_root:
             fail("Only the root module can use the 'gcc' extension")
 
+    DEFAULT_FEATURES = [
+        "minimal_warnings",
+        "strict_warnings",
+        "treat_warnings_as_errors",
+    ]
+
+
+    DEFAULT_WARNING_FLAGS = {
+        "minimal_warnings": [
+            "-Wall",
+            "-Wextra",
+            "-Wundef",
+            "-Wwrite-strings",
+            "-Wpointer-arith",
+            "-Wcast-align",
+            "-Wshadow",
+            "-Wswitch-bool",
+            "-Wredundant-decls",
+            "-Wswitch-enum",
+            "-Wtype-limits",
+            "-Wformat",
+            "-Wformat-security",
+            "-Wconversion",
+            "-Wlogical-op",
+            "-Wreturn-local-addr",
+            "-Wunused-but-set-variable",
+            "-Wunused-parameter",
+            "-Wunused-but-set-parameter",
+            "-Wunused-variable",
+            "-Wunused",
+            "-Wparentheses",
+            "-Wuninitialized",
+            "-Wsequence-point",
+            "-Wsign-compare",
+            "-Wignored-qualifiers",
+            "-Wcast-qual",
+            "-Wreturn-type",
+            "-Wcomment"
+        ],
+        "strict_warnings": [
```

stricts sound like only the nerds would apply that. I'd rename this to recommended.

```suggestion
        "recommended_warnings": [
```

---

I'm against renaming this. Maybe that outs me as a "nerd", but then I'm proud to be a safety-nerd. :wink: 

"strict_warnings" means that this is the strict set of compiler warnings that we as a project want to enforce.

In comparison "recommended" does not give you any knowledge why we recommend it.
Based on the scenario "recommended" may also have very different results.
For example I would recommend different compiler warnings for QM and ASIL-B code.

TLDR: "Recommended" is too ambiguous. Please stick with the current naming.

---

If "strict" is to be enforced, it should be moved to "minimal". Or do we want three groups, minimal + strict + recommended?